### PR TITLE
arrayfire-ml: init at 0.1.0

### PR DESCRIPTION
**Pull request by chessai**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @chessai

### CODE REVIEW

**Comment by chessai** [comment]
version is 0.1.0 as per the cmakefile.txt

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{ stdenv, fetchurl, fetchFromGitHub, cmake, pkgconfig, arrayfire }:
+
+let
+  version = "0.1.0";
+in stdenv.mkDerivation rec {
+  pname = "arrayfire-ml";
+  inherit version;
```
Comment (Lassulus) [inline pkgs/development/libraries/arrayfire-ml/default.nix]:
```suggestion
stdenv.mkDerivation rec {
  pname = "arrayfire-ml";
  version = "0.1.0";
```

Comment (Lassulus) [inline pkgs/development/libraries/arrayfire-ml/default.nix]: this should be:
```
stdenv.mkDerivation rec {
  pname = "arrayfire-ml";
  version = "0.1.0";
```

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    description = "C and C++ machine learning library built on top of the ArrayFire library";
+    longDescription = ''
+      ArrayFire ML is a C and C++ machine learning library built on top of the ArrayFire library.
+      This library leverages arrayfire's cross platform support to provide high performance machine learning algorithms for multi-core CPUs, NVIDIA and AMD GPUs, and other accelerators.
+    '';
+    license = licenses.bsd3;
+    homepage = "https://arrayfire.com/";
+    platforms = platforms.linux ++ platforms.darwin;
+    maintainers = with maintainers; [ chessai ];
+    inherit version;
```
Comment (Lassulus) [inline pkgs/development/libraries/arrayfire-ml/default.nix]: remove the version here, it's not needed in meta

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
+    mkdir -p $out/lib
+    mkdir -p $out/LICENSES
+    mkdir -p $out/share
+
+    cp libafml.so* $out/lib/.
+    cp -r ${src}/include $out/.
+    cp ${src}/LICENSE $out/LICENSES/BSD\ 3-Clause.txt
+    cp -r ${src}/examples $out/share/.
```
Comment (Lassulus) [inline pkgs/development/libraries/arrayfire-ml/default.nix]: you don't need to reference ${src} here, it's automatically extracted to the current working directory

**Review by Lassulus — CHANGES_REQUESTED** [review]

**Comment by chessai** [comment]
@Lassulus all suggestions applied. Rebased.

**Comment by Lassulus** [comment]
whupps, sorry for the delay, this now needs a rebase

**Comment by chessai** [comment]
rebased

**Comment by Lassulus** [comment]
getting this error now:
```
[1/6/9 built, 89 copied (821.1/821.4 MiB), 179.4 MiB DL] building arrayfire-3. [1/6/9 built, 89 copied (821.1/821.4 MiB), 179.4 MiB DL] building arrayfire-3. builder for '/nix/store/jq7cf2x27vd2a8w916jkbzdxjr1xj7by-arrayfire-ml-0.1.0.drv' failed with exit code 1; last 10 log lines:
  /build/source/examples/autograd.cpp:60:12: warning: format '%s' expects a matching 'char*' argument [-Wformat=]
       printf("%s:%d No Gradient check Failed\n");
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /build/source/examples/autograd.cpp:60:12: warning: format '%d' expects a matching 'int' argument [-Wformat=]
  [ 93%] Linking CXX executable xor
  [ 93%] Built target xor
  [100%] Linking CXX executable autograd
  [100%] Built target autograd
  installing
  cp: cannot stat 'include/': No such file or directory
cannot build derivation '/nix/store/w8mnr02l6frvf1dzixmr8r3yil2ab2k5-env.drv': 1 dependencies couldn't be built
```

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{ stdenv, fetchurl, fetchFromGitHub, cmake, pkgconfig, arrayfire }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrayfire-ml/default.nix:1]:
```suggestion
{ stdenv, fetchurl, fetchFromGitHub, cmake, pkg-config, arrayfire }:
```

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "arrayfire-ml";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "arrayfire";
+    repo = "arrayfire-ml";
+    rev = "fde97a5bf157936c8d904d0c857c6cb3275ee527";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrayfire-ml/default.nix:10]: Please track a tag.

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    sha256 = "0ij6g36zcxqgcyhcy28ycg7jf1hgw4p9hv5l3jmkzis0yqnxm346";
+  };
+
+  enableParallelBuilding = true;
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrayfire-ml/default.nix:15]:
```suggestion
```

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  nativeBuildInputs = [ cmake pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrayfire-ml/default.nix:16]:
```suggestion
  nativeBuildInputs = [ cmake pkg-config ];
```

File: pkgs/development/libraries/arrayfire-ml/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+  buildInputs = [ arrayfire ];
+
+  installPhase = ''
+    mkdir -p $out/lib
+    mkdir -p $out/LICENSES
+    mkdir -p $out/share
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrayfire-ml/default.nix:23]:
```suggestion
    mkdir -p $out/LICENSES $out/lib $out/share
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**